apps/prairielearn/src/question-servers/freeform.ts

```diff
@@ -72,16 +65,11 @@ interface QuestionProcessingContext {
 type ElementExtensionNameDirMap = Record<string, Record<string, ElementExtensionJsonExtension>>;
 type ElementNameMap = Record<
   string,
-  | (ElementCourseJson & {
-      name: string;
-      directory: string;
-      type: 'course';
-    })
-  | (ElementCoreJson & {
-      name: string;
-      directory: string;
-      type: 'core';
-    })
+  (ElementCourseJson | ElementCoreJson) & {
+    name: string;
+    directory: string;
+    type: 'course';
```

Comment:
nit: does this need to be `type: 'course' | 'core'`?

Reply:
I incorrectly tried to simplify this. Restored in ef9aa74.